File: Week 01/id_719/LeetCode_283_719.java
// 方法1： 先交换再补0
class LeetCode_283_719 {
    public void moveZeroes(int[] nums) {
        int j = 0;
        for (int num : nums) {
            if (num != 0) {
                nums[j] = num;
                j++;
            }
        }
        while (j < nums.length) {
            nums[j] = 0;
            j++;
        }
    }
}

File: Week 01/id_719/LeetCode_283_719.js
/**
 * 方法：指针i与j
 *   j指向0（除非i与j相同）
 *   i循环，当非0时，与j的指向值交换
 */

var moveZeroes = function(nums) {
    let j = 0;
    let { length } = nums;

    for (let i = 0; i < length; i++) {
        // 交换条件
        if (nums[i] !== 0) {
            nums[j] = nums[i];
            if (i !== j) {
                nums[i] = 0;
            }
            j++;
        }
    }
};